tests/Unit/TestHelpers.hpp

```diff
@@ -91,6 +91,8 @@ T serialize_and_deserialize(const T& t) {
 }
 
 /// Test for copy semantics assuming operator== is implement correctly
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wparentheses"
```

Comment:
True, but to make it work we actually would have to put the pragmas around every use of a CHECK-like macro, even the ones that don't actually cause a warning.  As I mentioned above, catch tries to disable the warning with a pragma.  Since most setups can't reproduce the warning, it apparently usually succeeds.  But it doesn't do a scoped disable like we do, but a global one, so it disables the warning for the rest of the compilation unit.

I see three options:
1. Ignore it, just say we won't worry about this warning in tests.
2. Actually wrap all 1000-and-counting uses of the macro in our tests
3. Write a wrapper macro that does the scoped disable and use that instead.

1 is the current state of this pull request and 2 or 3 seem out-of-scope of this change, so let's just merge this and then we can figure out what to do about the problem in general.